tests/framework/clients/rancher/v1/client.go

```diff
@@ -283,6 +287,45 @@ func (c *NamespacedSteveClient) Update(existing *SteveAPIObject, updates any) (*
 	return c.SteveClient.Update(existing, updates)
 }
 
+func (c *NamespacedSteveClient) PerformPutCaptureHeaders(host, token, endpoint, name string, payload interface{}) (http.Header, []byte, error) {
```

Comment:
I think instead of adding this method to the client, with the below ways, it might be possible to grab the header:

1. After the update call, from the returned Steve object: `mysteveObject.JSONResp["header"]` or `"headers"`, might give the headers
2. If the above doesn't work, the other way might be kind of halfway through usage by doing the update call like this: `client.Steve.Ops.DoUpdate()`, the parameter that's used to store the response object in the `DoUpdate` call might have the headers too

That way, we should be able to grab the headers and handle them in the tests, and we would prevent a new HTTP client instantiation

Reply:
Do you mean using client.Steve.Ops.DoUpdate() directly? Because that's what's called in `func (c *SteveClient) Update(existing *SteveAPIObject, updates any) (*SteveAPIObject, error)` and the same response is returned from DoUpdate, and there aren't headers returned

Reply:
So, after a bit of experiment. My suggestions relied on using this `respObject` parameter that we send to this below-linked function. My assumption was; that once the function is returned, the `respObject` would have the following headers:

https://github.com/rancher/rancher/blob/release/v2.8/tests/framework/pkg/clientbase/ops.go#L133

Resp here has the headers, but neither of my suggestions would work because we deserialize only the **body** in that variable.

https://github.com/rancher/rancher/blob/release/v2.8/tests/framework/pkg/clientbase/ops.go#L169. 

So, you can ignore/resolve my comment about this. Sorry! And thank you for your experiments, Andrew!

Reply:
Sounds good, thanks so much!